OloEngine/src/OloEngine/Serialization/MemoryLayout.h

```
@@ -70,86 +556,530 @@ namespace OloEngine
+            OloEngine::FTypeLayoutDesc::Initialize(TypeDesc); \
+        } \
+        return TypeDesc; }
+
+/**
+ * @def REGISTER_INLINE_TYPE_LAYOUT
+ * @brief Registers an inline type layout with the global registry
+ */
+#define REGISTER_INLINE_TYPE_LAYOUT(T) \
+    static struct ANONYMOUS_VARIABLE(RegisterTypeLayout) { \
+        ANONYMOUS_VARIABLE(RegisterTypeLayout)() { \
+            T::StaticGetTypeLayout().Name = TEXT(#T); \
```

Clang (14.0.6)

[warning] macro argument should be enclosed in parentheses

(bugprone-macro-parentheses)